### PR TITLE
Add wikipedia search

### PR DESCRIPTION
**Pull request by Zevleg**
Ticket [Implement wikipedia search](https://app.clickup.com/t/85ztpk31z)

image

Captura de pantalla 2023-08-04 a la(s) 2 18 22 p m

### CODE REVIEW

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/reasoning/WikipediaSearchExample.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import com.xebia.functional.gpt4all.getOrThrow
+import com.xebia.functional.xef.auto.ai
+import com.xebia.functional.xef.reasoning.wikipedia.WikipediaClient
+
+suspend fun main() {
+    ai {
+        val client = WikipediaClient()
+
+        val searchData = WikipediaClient.SearchData(
+            search = "Capital%20de%20Colombia"
```
Comment (raulraja) [inline examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/reasoning/WikipediaSearchExample.kt:12]: This should use URL encoding underneath as we do in #296

**Review by raulraja — APPROVED** [review]
Looks great @Zevleg ! Let's address the URL encoding so it encodes the search input automatically even if you send sentences with spaces

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/reasoning/WikipediaSearchExample.kt (hunk 1 above)
Comment (Zevleg) [reply examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/reasoning/WikipediaSearchExample.kt:12]: Done

File: reasoning/src/commonMain/kotlin/com/xebia/functional/xef/reasoning/wikipedia/WikipediaClient.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  data class Page(
+    @SerialName("pageid") val pageId: Int,
+    val title: String,
+    @SerialName("extract") val document: String
+  )
+
+  suspend fun search(searchData: SearchData): SearchResult {
+    return http
+      .get(
+        "https://en.wikipedia.org/w/api.php?format=json&action=query&list=search&srsearch=${searchData.search}"
```
Comment (raulraja) [inline reasoning/src/commonMain/kotlin/com/xebia/functional/xef/reasoning/wikipedia/WikipediaClient.kt:72]: I think `searchData.search` also needs to be URL encoded.

**Review by raulraja — APPROVED** [review]

File: reasoning/src/commonMain/kotlin/com/xebia/functional/xef/reasoning/wikipedia/WikipediaClient.kt (hunk 2 above)
Comment (Zevleg) [reply reasoning/src/commonMain/kotlin/com/xebia/functional/xef/reasoning/wikipedia/WikipediaClient.kt]: Done

**Review by raulraja — APPROVED** [review]
Thank you @Zevleg !

**Review by raulraja — APPROVED** [review]

**Review by raulraja — APPROVED** [review]

**Pull request merged**